src/passes/literalExpressionEvaluator/literalExpressionEvaluator.ts

```diff
@@ -265,9 +344,34 @@ function createLiteralFromType(typeString: string): RationalLiteral | null {
     const numerator = Number(numeratorString);
     const denominator = Number(denominatorString);
     if (!isNaN(numerator) && !isNaN(denominator)) {
-      return new RationalLiteral(BigInt(numeratorString), BigInt(denominatorString));
+      const result = generateInfo(
+        new RationalLiteral(BigInt(numeratorString), BigInt(denominatorString)),
+        node,
+        ast,
+      );
+      ast.replaceNode(node, result.node);
+      return result;
     }
   }
 
-  return null;
+  return generateInfo(null, node, ast);
+}
+
+function createNumberLiteralNode(result: RationalLiteral, ast: AST): Literal {
+  const intValue = result.toInteger();
+  if (intValue === null) {
+    throw new TranspileFailedError('Attempted to make node for non-integral literal');
+  }
+  return createNumberLiteral(intValue, ast);
+}
+
+function generateInfo(
+  value: RationalLiteral | boolean | null,
+  node: ASTNode,
+  ast: AST,
+): NodeEvaluationInfo {
+  if (value === null) return { evaluation: value, node: node };
+  else if (typeof value === 'boolean')
+    return { evaluation: value, node: createBoolLiteral(value, ast) };
+  else return { evaluation: value, node: createNumberLiteralNode(value, ast) };
```

Comment:
Just for the sake of extra security, I would add a condition for this last else statement `typeof value === number` and throw an error in other case